### PR TITLE
Support ssh_config ProxyCommand

### PR DESCRIPTION
**Pull request by lewisthompson**

### CODE REVIEW

**Comment by pahaz** [comment]
Interesting proposal. Could you please, describe the your use case in the README?

**Comment by lewisthompson** [comment]
Sure I'd be happy to. I'll sort something out a week from now as I'm
currently away on vacation.

**Comment by pahaz** [comment]
@lewisthompson, Hi! How your vacation? :) 
Do you plan on writing your use case in the README?

And thanks again for your PR.

**Comment by lewisthompson** [comment]
Hey @pahaz I've just updated the PR and included some documentation.  Essentially if there is a ProxyCommand present in the user's ssh_config file all SSH traffic will be routed through it, as per the way ssh(1) will do.  In this PR I've added an option to disable this functionality.  It is also possible to override (or specify) ssh_proxy.  I hope the documentation is self explanatory, if not, I tested with the following:

```python
ssh_proxy = paramiko.ProxyCommand('ping google.com')
tunnel = SSHTunnelForwarder(ssh_address_or_host=(ssh_server, ssh_port),
                            ssh_username=ssh_username,
                            ssh_password=ssh_password,
                            remote_bind_address=(remote_server, remote_port),
                            ssh_proxy_enabled=True,
                            ssh_proxy=ssh_proxy)
```

obviously this will time out reading/writing from stdin/stdout of `ping google.com` isn't going to give help us much.  Common examples include `nc` etc.

File: sshtunnel.py [hunk 1]
```diff
@@ -603,7 +610,12 @@ def __init__(
         ## CREATE THE TUNNELS
         self.tunnel_is_up = {}  # handle status of the other side of the tunnel
         try:
-            self._transport = paramiko.Transport((ssh_host, ssh_port))
+            if ssh_proxy and ssh_proxy_enabled:
+                self.logger.debug('Connecting with ProxyCommand {0}'
+                                  .format(repr(ssh_proxy.cmd)))
+                self._transport = paramiko.Transport(ssh_proxy)
+            else:
+                self._transport = paramiko.Transport(ssh_host, ssh_port)
```
Comment (pahaz) [inline sshtunnel.py]: `self._transport = paramiko.Transport` expect sock pair as first argument.

File: sshtunnel.py [hunk 2]
```diff
@@ -577,6 +581,9 @@ def __init__(
                 hostname_info.get('identityfile', [None])[0]
             ssh_port = ssh_port if ssh_port else \
                 hostname_info.get('port')
+            if not ssh_proxy and hostname_info.get('proxycommand'):
```
Comment (pahaz) [inline sshtunnel.py]: Please change code style: `ssh_proxy = ssh_proxy if ssh_proxy else hostname_info.get('proxycommand')`

**Comment by pahaz** [comment]
Great work! Can you code issuse and rebase from master?

**Comment by lewisthompson** [comment]
@pahaz Updated.  Apologies it took a while - was away once again.

**Pull request merged**